src/picologging/handlers.py

```diff
@@ -525,15 +526,62 @@ def __init__(self, queue):
         super().__init__()
         self.queue = queue
 
+    def enqueue(self, record):
+        """
+        Enqueue a record.
+
+        The base implementation uses put_nowait. You may want to override
+        this method if you want to use blocking, timeouts or custom queue
+        implementations.
+        """
+        self.queue.put_nowait(record)
+
+    def prepare(self, record):
+        """
+        Prepare a record for queuing. The object returned by this method is
+        enqueued.
+
+        The base implementation formats the record to merge the message and
+        arguments, and removes unpickleable items from the record in-place.
+        Specifically, it overwrites the record's `msg` and
+        `message` attributes with the merged message (obtained by
+        calling the handler's `format` method), and sets the `args`,
+        `exc_info` and `exc_text` attributes to None.
+
+        You might want to override this method if you want to convert
+        the record to a dict or JSON string, or send a modified copy
+        of the record while leaving the original intact.
+        """
+        # The format operation gets traceback text into record.exc_text
+        # (if there's exception data), and also returns the formatted
+        # message. We can then use this to replace the original
+        # msg + args, as these might be unpickleable. We also zap the
+        # exc_info, exc_text and stack_info attributes, as they are no longer
+        # needed and, if not None, will typically not be pickleable.
+        msg = self.format(record)
+        # bpo-35726: make copy of record to avoid affecting other handlers in the chain.
+        record = picologging.LogRecord(
```

Comment:
In CPython, this is record = copy.copy(record). However, our version of LogRecord *cannot* be copied, due to the logic of __reduce_ex__ in copyreg. So I manually copied it, basically. I will make an issue so we can discuss whether we want it to be copyable.

(Note that being able to copy something isn't always same as being able to pickle something! Similar code paths but not the same.)

Reply:
TODO: A test that checks the robustness of prepare